explorer/forms.py

```diff
@@ -49,8 +48,14 @@ def clean(self):
 
     @property
     def created_by_user_email(self):
-        return self.instance.created_by_user.email if \
-            self.instance.created_by_user else "--"
+        user = self.instance.created_by_user
```

Comment:
Might be neater without assigning the user first;

```
            try:
                return self.instance.created_by_user.email
            except AttributeError:
                return "--"
```

You could also set a property on your custom user called `email` that returned some way of identifying your users. That would avoid attribute errors with third party apps.

Reply:
Just thinking about this again, perhaps it makes more sense, if a user doesn't have an email address, just return the string representation of a user.

```
try:
    return self.instance.created_by_user.email
except AttributeError:
    return str(self.instance.created_by_user)
```

Then this gives projects that don't have an `email` field the chance to add an `email` property to do something custom, or just display the default representation of a user object.

Reply:
Yes, your solution is better. By the way, I'm just wondering why you are using the email and not using the string representation of the user in the first place?